Loop/Managers/DeviceDataManager.swift

```diff
@@ -548,13 +548,18 @@ final class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverD
                 return (quantity: $0.quantity, date: $0.startDate, isDisplayOnly: false)
             }
 
-            glucoseStore.addGlucoseValues(newGlucose, device: nil) { (_, _, error) -> Void in
-                if let error = error {
-                    self.logger.addError(error, fromSource: "GlucoseStore")
-                }
+            if (newGlucose.count > 0) {
 
-                NSNotificationCenter.defaultCenter().postNotificationName(self.dynamicType.GlucoseUpdatedNotification, object: self)
+                glucoseStore.addGlucoseValues(newGlucose, device: nil) { (_, _, error) -> Void in
```

Comment:
oh geez I had no idea this was returning an error. I'll put a guard statement in LoopKit.